InvalidContentTypeError on attempting PATCH: application/vnd.api+json

When upgrading from ember-cli@3.3.0 to 3.4.0, issuing `ember serve --proxy` returns InvalidContentTypeError every time I attempt to PATCH to the proxyed API. The initial GET appears to work fine. Since I have a working environment, I will proceed now to paste both working and broken outputs from specified commands and the `ember serve` output.


## Working
---
Output from `ember version --verbose && npm --version && yarn --version`:
```
$ ember version --verbose
ember-cli: 3.3.0
http_parser: 2.7.1
node: 8.10.0
v8: 6.2.414.50
uv: 1.18.0
zlib: 1.2.11
ares: 1.14.0
modules: 57
nghttp2: 1.30.0
openssl: 1.1.0g
icu: 60.2
unicode: 10.0
cldr: 32.0.1
tz: 2017c
os: linux x64

$ npm --version
3.5.2

$ yarn --version
zsh: command not found: yarn

```
```
$ ember serve --proxy=http://localhost:8080
Proxying to http://localhost:8080
/home/.../app/validations/dnsRecord.js
  3:2  error  'validateNumber' is defined but never used  no-unused-vars
✖ 1 problem (1 error, 0 warnings)
Build successful (3001ms) – Serving on http://localhost:4200/
Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
Package /assets/vendor.js (1)                 | 744ms
Rollup (1)                                    | 366ms
broccoli-persistent-filter:EslintValid... (3) | 273ms (91 ms)
Babel: lodash (1)                             | 197ms
broccoli-persistent-filter:SimpleReplace (3)  | 183ms (61 ms)
Funnel (31)                                   | 165ms (5 ms)
Funnel: Addon JS (35)                         | 151ms (4 ms)
GET /api/v1/main-model?include=relationship%2Crelationship.nested 200 30.154 ms - -
GET /api/v1/main-model/8 200 9.587 ms - 380
PATCH /api/v1/main-model/8 200 17.497 ms - 416
```

## Broken
---
```
$ ember version --verbose
WARNING: 
WARNING: /home/.../node_modules/ember-cli-broccoli-sane-watcher/node_modules/sane:
WARNING:    Missing dependency 'fsevents'
WARNING: 
WARNING: /home/.../node_modules/qunit/node_modules/sane:
WARNING:    Missing dependency 'fsevents'
WARNING: 
WARNING: /home/.../node_modules/sane:
WARNING:    Missing dependency 'fsevents'
WARNING: 
WARNING: /home/.../node_modules/simple-html-tokenizer/commands:
WARNING:    Missing package directory
ember-cli: 3.4.1
http_parser: 2.7.1
node: 8.10.0
v8: 6.2.414.50
uv: 1.18.0
zlib: 1.2.11
ares: 1.14.0
modules: 57
nghttp2: 1.30.0
openssl: 1.1.0g
icu: 60.2
unicode: 10.0
cldr: 32.0.1
tz: 2017c
os: linux x64

# npm and yarn status unchanged
```

```
WARNING:
WARNING: /home/.../node_modules/ember-cli-broccoli-sane-watcher/node_modules/sane:
WARNING:    Missing dependency 'fsevents'
WARNING:
WARNING: /home/.../node_modules/qunit/node_modules/sane:
WARNING:    Missing dependency 'fsevents'
WARNING:
WARNING: /home/.../node_modules/sane:
WARNING:    Missing dependency 'fsevents'
WARNING:
WARNING: /home/.../node_modules/simple-html-tokenizer/commands:
WARNING:    Missing package directory
DEPRECATION: Ember will stop including jQuery by default in an upcoming version. If you wish keep using jQuery in your application explicitly add `@ember/jquery` to your package.json
Proxying to http://localhost:8080
/home/.../app/validations/dnsRecord.js
  3:2  error  'validateNumber' is defined but never used  no-unused-vars
✖ 1 problem (1 error, 0 warnings)
Build successful (17102ms) – Serving on http://localhost:4200/
Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
Babel: lodash (1)                             | 3662ms
Babel: ember-data (2)                         | 2529ms (1264 ms)
Babel: ember-test-helpers (2)                 | 1527ms (763 ms)
Babel: cdn-ui (2)                             | 1439ms (719 ms)
GET /api/v1/main-model?include=relationship%2Crelationship.nested 200 31.385 ms - -
GET /api/v1/main-model/8 200 14.955 ms - 381
InvalidContentTypeError: Could not parse content type header: application/vnd.api+json
    at createError (/home/.../node_modules/error/typed.js:31:22)
    at anyBody (/home/.../node_modules/body/any.js:36:18)
    at Server.handler (/home/.../node_modules/tiny-lr/src/server.js:165:32)
    at Layer.handle [as handle_request] (/home/.../node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/.../node_modules/express/lib/router/index.js:317:13)
    at /home/.../node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/home/.../node_modules/express/lib/router/index.js:335:12)
    at next (/home/.../node_modules/express/lib/router/index.js:275:10)
    at compression (/home/.../node_modules/compression/index.js:220:5)
    at Layer.handle [as handle_request] (/home/.../node_modules/express/lib/router/layer.js:95:5)
```

This is shown in Firefox's console:
```
XHR PATCH http://localhost:4200/api/v1/main-model/8
[HTTP/1.1 415 Unsupported Media Type 9ms]
Error: Ember Data Request PATCH /api/v1/main-model/8 returned a 415
Payload (text/html; charset=utf-8)
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>InvalidContentTypeError: Could not parse content type header: application/vnd.api+json<br> &nbsp; &nbsp;at createError (/home/.../node_modules/error/typed.js:31:22)<br> &nbsp; &nbsp;at anyBody (/home/.../node_modules/body/any.js:36:18)<br> &nbsp; &nbsp;at Server.handler (/home/.../node_modules/tiny-lr/src/server.js:165:32)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/home/.../node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/home/.../node_modules/express/lib/router/index.js:317:13)<br> &nbsp; &nbsp;at /home/.../node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/home/.../node_modules/express/lib/router/index.js:335:12)<br> &nbsp; &nbsp;at next (/home/.../node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at compression (/home/.../node_modules/compression/index.js:220:5)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/home/.../node_modules/express/lib/router/layer.js:95:5)</pre>
</body>
</html>
```

--- a/lib/tasks/server/livereload-server.js
+++ b/lib/tasks/server/livereload-server.js
@@ -7,6 +7,7 @@ const FSTree = require('fs-tree-diff');
 const logger = require('heimdalljs-logger')('ember-cli:live-reload:');
 const fs = require('fs');
 const Promise = require('rsvp').Promise;
+const cleanBaseUrl = require('clean-base-url');
 
 function isNotRemoved(entryTuple) {
   let operation = entryTuple[0];
@@ -74,10 +75,11 @@ module.exports = class LiveReloadServer {
         this.liveReloadServer = new Server({
           app: this.app,
           dashboard: 'false',
-          prefix: options.liveReloadPrefix,
+          prefix: '/',
           port: options.port,
         });
       }
+      this.liveReloadPrefix = cleanBaseUrl(options.liveReloadPrefix);
       this.start();
     } else {
       this.ui.writeWarnLine('Livereload server manually disabled.');
@@ -103,7 +105,7 @@ module.exports = class LiveReloadServer {
     this.httpServer.on('upgrade', this.liveReloadServer.websocketify.bind(this.liveReloadServer));
     this.httpServer.on('error', this.liveReloadServer.error.bind(this.liveReloadServer));
     this.httpServer.on('close', this.liveReloadServer.close.bind(this.liveReloadServer));
-    this.app.use(this.liveReloadServer.handler.bind(this.liveReloadServer));
+    this.app.use(this.liveReloadPrefix, this.liveReloadServer.handler.bind(this.liveReloadServer));
   }
 
   createHttpsServer(options, Server) {

--- a/lib/tasks/server/livereload-server.js
+++ b/lib/tasks/server/livereload-server.js
@@ -43,6 +43,8 @@ function readSSLandCert(sslKey, sslCert) {
   return { key, cert };
 }
 
+const DEFAULT_PREFIX = '/';
+
 module.exports = class LiveReloadServer {
   constructor({ app, watcher, ui, project, analytics, httpServer }) {
     this.app = app;
@@ -51,11 +53,16 @@ module.exports = class LiveReloadServer {
     this.project = project;
     this.analytics = analytics;
     this.httpServer = httpServer;
+    this.liveReloadPrefix = DEFAULT_PREFIX;
   }
 
   setupMiddleware(options) {
     const tinylr = require('tiny-lr');
     const Server = tinylr.Server;
+
+    if (options.liveReloadPrefix) {
+      this.liveReloadPrefix = cleanBaseUrl(options.liveReloadPrefix);
+    }
 
     if (options.liveReload) {
       if (options.liveReloadPort && options.port !== options.liveReloadPort) {
@@ -75,11 +82,10 @@ module.exports = class LiveReloadServer {
         this.liveReloadServer = new Server({
           app: this.app,
           dashboard: 'false',
-          prefix: '/',
+          prefix: DEFAULT_PREFIX,
           port: options.port,
         });
       }
-      this.liveReloadPrefix = cleanBaseUrl(options.liveReloadPrefix);
       this.start();
     } else {
       this.ui.writeWarnLine('Livereload server manually disabled.');

--- a/tests/unit/tasks/server/livereload-server-test.js
+++ b/tests/unit/tasks/server/livereload-server-test.js
@@ -589,9 +589,11 @@ describe('livereload-server', function() {
 
     describe('specific files', function() {
       let reloadedFiles;
+      let changedOptions;
 
       let stubbedChanged = function(options) {
         reloadedFiles = options.body.files;
+        changedOptions = options;
       };
 
       beforeEach(function() {
@@ -655,13 +657,6 @@ describe('livereload-server', function() {
       });
 
       it('triggers livereload with "LiveReload files" if no results.directory was provided', function() {
-        let changedOptions;
-        subject.liveReloadServer = {
-          changed(options) {
-            changedOptions = options;
-          },
-        };
-
         subject.didChange({});
 
         expect(changedOptions).to.deep.equal({

